src/Ormolu/Printer/Meat/Declaration/Deriving.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+
+module Ormolu.Printer.Meat.Declaration.Deriving
+  ( p_derivDecl
+  )
+where
+
+import GHC
+import Ormolu.Printer.Combinators
+import Ormolu.Printer.Meat.Type
+import Ormolu.Utils
+
+p_derivDecl :: DerivDecl GhcPs -> R ()
+p_derivDecl = \case
+  d@DerivDecl {} -> p_standaloneDerivDecl d
+  XDerivDecl _ -> notImplemented "XDerivDecl standalone deriving"
+
+p_standaloneDerivDecl :: DerivDecl GhcPs -> R ()
+p_standaloneDerivDecl DerivDecl {..} = line $ do
+    txt "deriving"
+    case deriv_strategy of
+      Nothing ->
+        breakpoint
+      Just l -> locatedVia Nothing l $ \case
```

Comment:
I don't understand how it renders the other cases given that this case expression doesn't implement some matches? How is it capable of rendering "deriving via" instances for example? Is `via` included in the `deriv_type` somehow? Once we (or well, I) understand this, it'll be easier to tell whether or not we can re-use existing rendering logic for `DerivStrategy`.

If `located (hsib_body (hswc_body deriv_type)) p_hsType` is too crude I propose we infer approximately the span that includes everything by using the `combineSrcSpans'` function. It works like this: combine spans of individual elements and then feed that to `switchLayout` that should be put around the entire thing I guess. Here is an example:

https://github.com/sphaso/ormolu/blob/feature%2F50_standalone_deriving/src/Ormolu/Printer/Meat/Declaration/Data.hs#L51

Another important thing that is missing is handling of `deriv_overlap_mode`. Overlapping modes should be handled similar to how it's done for instance declarations:

https://github.com/sphaso/ormolu/blob/feature%2F50_standalone_deriving/src/Ormolu/Printer/Meat/Declaration/Instance.hs#L33

Perhaps some part of it could be moved to a standalone function and reused.